### PR TITLE
Upgrade to DAML SDK 0.13.51

### PR DESCRIPTION
**Pull request by fabiotudone-da**
This also aligns `ExampleServer` with 0.13.51's `ReferenceServer`, which has the additional advantage of being much simpler code and thus makes for a better example.

I had to upgrade anyway in order to test that [this PR](https://github.com/digital-asset/daml/pull/4388) (on a 0.13.51 codebase) fixes the issue it is meant to fix.

### CODE REVIEW

**Comment by SamirTalwar-DA** [comment]
If you don't need the `extraParticipants` logic in this repository, you can just use `kvutils.app.Runner` and `ProgramResource` to really knock down the lines of code.

**Comment by dasormeter** [comment]
> If you don't need the `extraParticipants` logic in this repository, you can just use `kvutils.app.Runner` and `ProgramResource` to really knock down the lines of code.

actually that raises an interesting question @mziolekda  and @fabiotudone-da : what is the simplest but most representative way of spinning up multiple participants with the example server?  is it worth splitting out the write and read processes?  how many separate processes would we want to run?

similar conversations have come up with BTP about sawtooth / qldb / besu about having a single writer but multiple readers for each of the participants

**Comment by fabiotudone-da** [comment]
@SamirTalwar-DA @dasormeter Thanks for the feedback; I think the original example (pre-PR) didn't cover extra participants and I'll double-check but if so I'd rather follow Samir's suggestion and knock down those extra lines (in line with this PR's purpose, i.e. just upgrade).

I think having an example of multiple participants should be tracked though; what's a good way, GH issue perhaps?

**Comment by dasormeter** [comment]
> @SamirTalwar-DA @dasormeter Thanks for the feedback; I think the original example (pre-PR) didn't cover extra participants and I'll double-check but if so I'd rather follow Samir's suggestion and knock down those extra lines (in line with this PR's purpose, i.e. just upgrade).
> 
> I think having an example of multiple participants should be tracked though; what's a good way, GH issue perhaps?

yeah think either jira issue or github issue is valid : probably jira better if it is expected to be an action for KV team, github if for other teams

File: build.sbt [hunk 1]
```diff
@@ -20,6 +19,12 @@ assemblyMergeStrategy in assembly := {
     MergeStrategy.first
   case PathList(ps @ _*) if ps.last startsWith "com/fasterxml/jackson" =>
     MergeStrategy.first
+  case PathList("google", "protobuf", n) if n endsWith ".proto" =>
+    // Both in protobuf and akka
+    MergeStrategy.first
+  case "module-info.class" =>
+    // In all 2.10 Jackson JARs
+    MergeStrategy.discard
```
Comment (dasormeter) [inline build.sbt]: @fabiotudone-da while there : do you know if we definitively need this and the other jackson merge strategy explicit definition?

Comment (fabiotudone-da) [reply build.sbt]: Unfortunately I think we do; in this case we're building an uberjar so we're merging all the JARs together and the [new Jackson version used by the newer SDKs support Java 9+ modules](https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.10#jdk-9-module-info) so every JAR includes the module declaration and they end up in the same classpath position. Similarly for google protobuf definitions that are provided by Akka as well.

This is due to [`sbt-assembly`'s design & defaults](https://github.com/sbt/sbt-assembly#merge-strategy): they are geared towards explicit resolution. It's not a bad thing per se as that implies more predictable builds, still it's kind of harder to use as one is required to examine, think about and resolve explicitly every clash.

**Comment by fabiotudone-da** [comment]
As discussed, will upgrade to 0.13.52 instead, in order to be able to use `kvutils.app.Runner`.

@miklos-da has a point in saying that, after this upgrade, we could just move to the main daml repo the diff w.r.t. this one (i.e. example docs + convenience cmdline options to manually run a participant uberjar) and not need to maintain this one anymore (or use it for other purposes).

**Comment by fabiotudone-da** [comment]
@SamirTalwar-DA Actually this example server _does_ support extra participants so I'd leave it as it is (and no need to upgrade to 0.13.52).

**Pull request merged**